src/Encoder/ImageEncoder.php

```diff
@@ -4,8 +4,9 @@
 
 namespace AM\InterventionRequest\Encoder;
 
-use Intervention\Image\Exception\NotWritableException;
-use Intervention\Image\Image;
+use Intervention\Image\Exceptions\NotWritableException;
+use Intervention\Image\Interfaces\EncodedImageInterface;
+use Intervention\Image\Interfaces\ImageInterface;
 
 class ImageEncoder
```

Comment:
```suggestion
final class ImageEncoder implements ImageEncoderInterface
```

Create an interface with this class. And use the interface as dependency in your services constructors